test_case/ERA5TimeSeriesDataset.py

```diff
@@ -138,14 +262,55 @@ def __getitem__(self, index):
     end_year = 2010
     input_vars = ['t2m', 'V500', 'U500', 'T500', 'Z500', 'Q500']
 
-    train_dataset = ERA5Dataset(data_path, start_year, end_year, input_vars=input_vars)
-    print (train_dataset)
-    forecast_step = 1
-    ds_x , ds_y = train_dataset.fetch_timeseries(forecast_step=1)
+    use_dali: bool = True
+    if not use_dali:
+        train_dataset = ERA5Dataset(
+            data_path, start_year, end_year, input_vars=input_vars
+        )
+        print(train_dataset)
+        forecast_step = 1
+        ds_x, ds_y = train_dataset.fetch_timeseries(forecast_step=1)
+
+        print(ds_x)
+
+        # or use with data loader for pytorch as follows:
+        # pytorch_dataset = PyTorchERA5Dataset(train_dataset)
+        # data_loader = DataLoader(pytorch_dataset, batch_size=16, shuffle=True)
+        # etc....
+
+    else:  # use_dali is True
+        pipe = dali.Pipeline(
+            batch_size=16,
+            num_threads=4,
+            prefetch_queue_depth=4,
+            py_num_workers=4,
+            device_id=0,
+            py_start_method="fork",
+        )
+        with pipe:
+            # Zarr source
+            source = SeqZarrSource()
+
+            # Update length of dataset
+            # self.total_length: int = len(source) // self.batch_size
+
+            # Read current batch
+            data = dali.fn.external_source(
+                source,
+                num_outputs=2,  # len(self.pipe_outputs),
+                parallel=True,
+                batch=True,
+                prefetch_queue_depth=4,
+                device="cpu",
+            )
+
+            # if self.device.type == "cuda":
+            # Move tensors to GPU as external_source won't do that
+            data = [d.gpu() for d in data]
 
-    print(ds_x)
+            # Set outputs
+            pipe.set_outputs(*data)
 
-    # or use with data loader for pytorch as follows:
-    # pytorch_dataset = PyTorchERA5Dataset(train_dataset)
-    # data_loader = DataLoader(pytorch_dataset, batch_size=16, shuffle=True)
-    # etc....
+        pipe.build()
+        arrays = pipe.run()
+        print(arrays)
```

Comment:
For reference, this should print:

```
(TensorListGPU(
    dtype=DALIDataType.FLOAT,
    num_samples=1,
    shape=[(6, 640, 1280)]), TensorListGPU(
    dtype=DALIDataType.FLOAT,
    num_samples=1,
    shape=[(6, 640, 1280)]))
```
i.e. the before and after time slices for the 6 ERA5 variables.